huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go

```diff
@@ -0,0 +1,847 @@
+	)
+	createFirewallClient, err := cfg.NewServiceClient(createFirewallProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW client: %s", err)
+	}
+
+	createFirewallPath := createFirewallClient.Endpoint + createFirewallHttpUrl
+	createFirewallPath = strings.ReplaceAll(createFirewallPath, "{project_id}", createFirewallClient.ProjectID)
+
+	createFirewallOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	createFirewallOpt.JSONBody = utils.RemoveNil(buildCreateFirewallBodyParams(d, cfg))
+	createFirewallResp, err := createFirewallClient.Request("POST", createFirewallPath, &createFirewallOpt)
+	if err != nil {
+		return diag.Errorf("error creating Firewall: %s", err)
+	}
+
+	createFirewallRespBody, err := utils.FlattenResponse(createFirewallResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	orderID := utils.PathSearch("order_id", createFirewallRespBody, "")
+	jobID := utils.PathSearch("job_id", createFirewallRespBody, "")
+
+	bssClient, err := cfg.BssV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating BSS v2 client: %s", err)
+	}
+
+	if orderID != "" {
+		err = common.WaitOrderComplete(ctx, bssClient, orderID.(string), d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		resourceId, err := common.WaitOrderResourceComplete(ctx, bssClient, orderID.(string), d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		d.SetId(resourceId)
+	} else {
+		d.SetId(jobID.(string))
+
+		err = createFirewallWaitingForStateCompleted(ctx, d, meta, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.Errorf("error waiting for the Firewall (%s) creation to complete: %s", d.Id(), err)
+		}
+	}
+
+	if _, ok := d.GetOk("east_west_firewall_inspection_cidr"); ok {
+		// create east west firewall
+		var (
+			createEastWestFirewallHttpUrl = "v1/{project_id}/firewall/east-west"
+		)
+
+		createEastWestFirewallPath := createFirewallClient.Endpoint + createEastWestFirewallHttpUrl
+		createEastWestFirewallPath = strings.ReplaceAll(createEastWestFirewallPath, "{project_id}", createFirewallClient.ProjectID)
+		createEastWestFirewallPath += fmt.Sprintf("?fw_instance_id=%s", d.Id())
+
+		createEastWestFirewallOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				200,
+			},
+			MoreHeaders: map[string]string{"Content-Type": "application/json"},
+		}
+
+		createEastWestFirewallOpt.JSONBody = utils.RemoveNil(buildCreateEastWestFirewallBodyParams(d))
+		createEastWestFirewallResp, err := createFirewallClient.Request("POST", createEastWestFirewallPath, &createEastWestFirewallOpt)
+		if err != nil {
+			return diag.Errorf("error creating east-west Firewall: %s", err)
+		}
+
+		_, err = utils.FlattenResponse(createEastWestFirewallResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
```

Comment:
It seems that there is no need to do this verification.

Reply:
fixed